### DISCUSSION TITLE
Scaler Pro Production Branches Cluster Size

### DISCUSSION THREAD

**Discussion by martin-opensky** [planetscale/discussion#615] [Q&A] [answered]
Hello,

I am looking to migrate some MySQL Databases from a privately managed VM to PlanetScale but need some help understanding Production branches and associated costs.

We have several DBs with the same schema, where customers need their data separated from other companies.  When migrating to PlanetScale I am looking at having a production branch for each separate customer in the same way we currently have separate DBs.  Looking at the docs it states: _Additional production branches are billed at the cost of your selected cluster size per month_

My question is: can we set different cluster sizes for each production branch or do all production branches require the identical cluster size?  Some DBs require less resources than others.

Could anyone share any experiences you have with a similar setup?

Thanks for your help
Martin

**Comment by mscoutermarsh** [comment, marked as answer]
Hi @martin-opensky,

Yes, you can set a different cluster size for each production branch. You can also quickly change between cluster sizes without downtime. You are only billed for the time that the branch is running at that size.

**Comment by martin-opensky** [comment]
Thank you @mscoutermarsh.

I have a follow up question: when there are multiple production branches can the data migration / database import tool be used to import data from the separate company databases into the separate production branches?

**Reply by mscoutermarsh** [reply]
The import tool currently only works once per database. Which could be another option, having each company in their own database instead of branch.

**Comment by martin-opensky** [comment]
Thank you @mscoutermarsh.

It seems like having a separate database instead of production branches would be a better approach.

I have a follow up to this.  In the Scaler Pro plan it states: _Deployed across 3 availability zones._
What does this mean when I have chosen eu-west-2 for my region?  Will the DB be available in other regions to eu-west-2?

Finally it also states: _1 production branch included with 2 replicas for high-availability_
Are the 2 replicas read-only and is it preferable to use a different connection when writing to the DB and reading from the DB?

**Reply by mscoutermarsh** [reply]
Think of an Availability Zone as a "building within the region". Each with separate power & connectivity, in case something happens to one building, we still have a copy of your database running in the 2 others. The primary and replicas are all still in the same region, yet isolated.

If you wanted the DB available for reads in a separate region (for latency purposes), you could add a read-only replica in a different region.

> Are the 2 replicas read-only and is it preferable to use a different connection when writing to the DB and reading from the DB?

Yes, they are available for reads. It's best to use separate connections when reading from replicas. Primarily I like this because it makes your code explicit about where the data is coming from. Since it is possible a replica might lag behind the primary.

We have some more info here: https://planetscale.com/docs/concepts/replicas#data-consistency-and-replication-lag. I wrote most of this doc, so if you have any more questions, I can help.

**Comment by martin-opensky** [comment]
Thank you @mscoutermarsh!
